Tests/appstoreconnect-cliTests/Operations/ReadPreReleaseVersionOperationTests.swift

```diff
@@ -0,0 +1,196 @@
+// Copyright 2020 Itty Bitty Apps Pty Ltd
+
+@testable import AppStoreConnectCLI
+import AppStoreConnect_Swift_SDK
+import Combine
+import XCTest
+
+final class ReadPreReleaseVersionOperationTests: XCTestCase {
+    typealias Operation = ReadPreReleaseVersionOperation
+    typealias Options = Operation.Options
+    typealias OperationError = ReadPreReleaseVersionOperation.Error
+
+    let successResponseRequestor = OneEndpointTestRequestor(response: { _ in
+        Future({ $0(.success(onePreRealeseVersionResponse)) }) }
+    )
+
+    let noResponseRequestor = OneEndpointTestRequestor(response: { _ in
+        Future{ $0(.success(noPreReleaseVersionResponse)) }}
+    )
+
+    let notUniqueRequestor = OneEndpointTestRequestor(response: { _ in
+        Future{ $0(.success(notUniqueResponse)) }}
+    )
+
+    func testOnePreReleaseVersion() throws {
+        let operation = Operation(options: Options(filterAppId: "1504341572", filterVersion: "1.0"))
+
+        let result = Result {
+            try operation.execute(with: successResponseRequestor).await()
+        }
+
+        switch result {
+        case .success(let output):
+            XCTAssertEqual(output.preReleaseVersion.attributes?.version, "1.0")
+        default:
+            XCTFail("Error in getting prereleaseVersion response")
+        }
```

Comment:
Tests that `throw` will fail correctly.  So you don't need to wrap the `await()` in a `Result`.

```suggestion
        let operation = Operation(options: Options(filterAppId: "1504341572", filterVersion: "1.0"))

        let output =  try operation.execute(with: successResponseRequestor).await()
        XCTAssertEqual(output.preReleaseVersion.attributes?.version, "1.0")
```